cmd/subctl/recover_broker_info.go

```diff
@@ -63,114 +55,57 @@ func init() {
 	rootCmd.AddCommand(recoverBrokerInfo)
 }
 
-func recoverBrokerFromConfigContext(brokerCluster *cluster.Info, brokerNamespace string, status reporter.Interface) error {
-	brokerObj, err := getBroker(brokerCluster, brokerNamespace, "Please try again with another cluster", status)
-	if err != nil {
-		return err
-	}
+func recoverBrokerInfoFromSubm(submCluster *cluster.Info, namespace string, status reporter.Interface) error {
+	brokerNamespace := submCluster.Submariner.Spec.BrokerK8sRemoteNamespace
+	brokerRestConfig := submCluster.RestConfig
 
-	clusters, err := brokerCluster.GetClusters(brokerNamespace)
+	status.Start("Checking if the Broker is installed on the Submariner cluster %q in namespace %q", submCluster.Name, brokerNamespace)
+
+	brokerObj, err := getBroker(brokerRestConfig, brokerNamespace)
 	if err != nil {
-		return status.Error(err, "error listing joined clusters")
+		status.Warning("Error getting Broker: %s", err)
```

Comment:
An error should be logged as such and abort the operation.

```suggestion
		return status.Error(err, "")
```

Reply:
We don't want to return here. This addresses the case where Broker is not installed on the current context and hence we move ahead to getting remote Brokerconfig and get BrokerObj from that remote cluster.

Reply:
Changed the return err in L103 to be able to return here.

Reply:
yeah this goes along with my comment in `getBroker` - I should've been clear about that. Basically we need to distinguish between "not found" and an unrecoverable error and, for the latter, abort. You could also just have `getBroker` propagate the error as is and check `apierrors.IsNotFound` here.